lib/tasks/certs.rake

```diff
@@ -202,4 +204,57 @@ namespace :certs do
       end
     end
   end
+
+
+
+  desc 'Check that LG certificate bundle matches certificates in certificate path'
+  task check_certificate_bundle: :environment do |t, args|
+    CertificateStore.instance.load_certs!(dir: 'config/certs')
+    bundled_certs = []
+    cert_bundle_file = File.read(IdentityConfig.store.login_certificate_bundle_file)
+    cert_bundle = cert_bundle_file.split(CertificateStore::END_CERTIFICATE).map do |cert|
+      cert += CertificateStore::END_CERTIFICATE
+      cert = Certificate.new(OpenSSL::X509::Certificate.new(cert))
+    end
+
+    if cert_bundle.map(&:sha1_fingerprint).sort != CertificateStore.instance.certificates.map(&:sha1_fingerprint).sort
+      puts <<-ERROR
+        #{IdentityConfig.store.login_certificate_bundle_file} does not match the certificates in #{IdentityConfig.store.certificate_store_directory}
+        Please run:
+        rake certs:generate_certificate_bundle
+      ERROR
+      exit 1
+    end
+  end
+
+  task generate_certificate_bundles: :environment do |t, args|
+    CertificateStore.instance.load_certs!(dir: 'config/certs')
+    File.write(
+      IdentityConfig.store.login_certificate_bundle_file,
+      CertificateStore.instance.certificates.sort_by(&:sha1_fingerprint).map(&:to_pem).join,
+    )
+
+    ficam_uri = URI('https://raw.githubusercontent.com/GSA/ficam-playbooks/staging/_fpki/tools/CACertificatesValidatingToFederalCommonPolicyG2.p7b')
+
+    response = Net::HTTP.get_response(ficam_uri)
+    body = response.body.force_encoding('UTF-8')
+    stdout, stderr, status = Open3.capture3('openssl', 'pkcs7', '-print_certs', '-inform', 'DER', stdin_data: body)
+    raw_certificates = stdout.strip
+
+    certificates = raw_certificates.split(CertificateStore::END_CERTIFICATE).map do |cert|
+      cert += CertificateStore::END_CERTIFICATE
+      cert = Certificate.new(OpenSSL::X509::Certificate.new(cert))
+    end
+
+    # Remove all certificates that are non-root cert and sign the Federal Bridge CA G4 cert
+    certificates.reject! do |x|
+      (x.key_id == '79:F0:00:49:EB:7F:77:C2:5D:41:02:65:34:8A:90:23:9B:1E:07:6F' &&
```

Comment:
WDYT of naming this as like `root_cert_key_id` or something?

Reply:
good call, thank you.  renamed to `federal_brige_ca_g4_key_id` in 7f59652

Reply:
oh one more thought, should we drop a comment with the date that cert expires? or do key-ids stay the same across updated certs

Reply:
My understanding is CAs can sign a new cert with the same key ID to replace them.  This one is signed by the common policy cert, so it's central to a lot of the other intermediate certs and I'm guessing when it expires there'll be some heads up before it expires in 2029.  That said, I added the expiration time with the caveat that we may have to change the key ID just so we have a note about that.